stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -856,3 +875,49 @@ for func in (:log, :sqrt)
         end
     end
 end
+
+"""
+    hermitianpart(A, uplo=:U) -> Hermitian
+
+Return the Hermitian part of the square matrix `A`, defined as `(A + A') / 2`, as a
+[`Hermitian`](@ref) matrix. For real matrices `A`, this is also known as the symmetric part
+of `A`. The optional argument `uplo` controls the corresponding argument of the
+[`Hermitian`](@ref) view. For real matrices, the latter is equivalent to a
+[`Symmetric`](@ref) view.
+
+See also [`hermitianpart!`](@ref) for the corresponding in-place operation.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+hermitianpart(A::AbstractMatrix{T}, uplo::Symbol=:U) where {T} =
+    hermitianpart!(copy_similar(A, Base.promote_op(/, T, Int)), uplo)
+hermitianpart(x::Number, uplo::Symbol=:U) = real(x)
```

Comment:
Seems to me we'd probably want `typeof(hermitianpart(x)) == eltype(hermitianpart([x;;]))` for `x::Number`.

Reply:
Thanks. Included in [94e5cee](https://github.com/JuliaLang/julia/pull/31836/commits/94e5ceea9f3f2ba7b84a3473da50d15473e81db0).